tests/e2e/050_nat_test.go

```diff
@@ -0,0 +1,516 @@
+					{
+						LocalIp:   linuxTap2IP,
+						LocalPort: tcpSvcLocalPort,
+					},
+				},
+			},
+		},
+	}
+
+	showUDPSvc := fmt.Sprintf("udp local %s:%d external %s:%d vrf 0  out2in-only",
+		linuxTap2IP, udpSvcLocalPort, udpSvcExtIP, udpSvcExtPort)
+	udpSvc := &vpp_nat.DNat44{
+		Label: udpSvcLabel,
+		StMappings: []*vpp_nat.DNat44_StaticMapping{
+			{
+				ExternalIp:   udpSvcExtIP,
+				ExternalPort: udpSvcExtPort,
+				Protocol:     vpp_nat.DNat44_UDP,
+				LocalIps: []*vpp_nat.DNat44_StaticMapping_LocalIP{
+					{
+						LocalIp:   linuxTap2IP,
+						LocalPort: udpSvcLocalPort,
+					},
+				},
+			},
+		},
+	}
+
+	// without proxy ARP, the client running from microservice1 (public net),
+	// will not get any response for ARP requests concerning service external IPs
+	// from VPP
+	svcExtIPsProxyArp := &vpp_l3.ProxyARP{
+		Interfaces: []*vpp_l3.ProxyARP_Interface{
+			{
+				Name: vppTap1Name,
+			},
+		},
+		Ranges: []*vpp_l3.ProxyARP_Range{
+			{
+				FirstIpAddr: tcpSvcExtIP,
+				LastIpAddr:  tcpSvcExtIP,
+			},
+			{
+				FirstIpAddr: udpSvcExtIP,
+				LastIpAddr:  udpSvcExtIP,
+			},
+		},
+	}
+
+	checkConfig := func(isConfigured bool) {
+		substringMatcher := func(substring string, negative bool) types.GomegaMatcher {
+			if negative {
+				return Not(ContainSubstring(substring))
+			} else {
+				return ContainSubstring(substring)
+			}
+		}
+		stdout, err := ctx.execVppctl("show", "nat44", "static", "mappings")
+		Expect(err).ToNot(HaveOccurred(), "Running `vppctl show nat44 addresses` failed")
+		Expect(stdout).To(substringMatcher(showTCPSvc, !isConfigured),
+			"Unexpected TCP static mapping configuration",
+		)
+		Expect(stdout).To(substringMatcher(showUDPSvc, !isConfigured),
+			"Unexpected UDP static mapping configuration",
+		)
+	}
+
+	checkConn := func(shouldSucceed bool) {
+		expected := Succeed()
+		if !shouldSucceed {
+			expected = Not(Succeed())
+		}
+
+		Expect(ctx.testConnection(ms1Name, ms2Name, tcpSvcExtIP, linuxTap2IP,
+			tcpSvcExtPort, tcpSvcLocalPort, false, tapv2InputNode)).To(expected)
+		Expect(ctx.testConnection(ms1Name, ms2Name, udpSvcExtIP, linuxTap2IP,
+			udpSvcExtPort, udpSvcLocalPort, true, tapv2InputNode)).To(expected)
+	}
+
+	ctx.startMicroservice(ms1Name)
+	ctx.startMicroservice(ms2Name)
+	checkConfig(false)
+	req := ctx.grpcClient.ChangeRequest()
+	err := req.Update(
+		vppTap1,
+		linuxTap1,
+		vppTap2,
+		linuxTap2,
+		ms2DefaultRoute,
+		svcExtIPsProxyArp,
+		natGlobal,
+		tcpSvc, udpSvc,
+	).Send(context.Background())
+	Expect(err).ToNot(HaveOccurred(), "Transaction creating public and private networks failed")
+
+	Eventually(ctx.getValueStateClb(vppTap1)).Should(Equal(kvs.ValueState_CONFIGURED),
+		"TAP attached to a newly started microservice1 should be eventually configured")
+	Eventually(ctx.getValueStateClb(vppTap2)).Should(Equal(kvs.ValueState_CONFIGURED),
+		"TAP attached to a newly started microservice2 should be eventually configured")
+
+	// check configuration
+	checkConfig(true)
```

Comment:
This assertion does not say anything to readers a has useless comment. 

Instead of `checkConfig` we could have 3 small helper functions:

```go
staticMappings := func() (string, error) {
	return ctx.execVppctl("show nat44 static mappings")
)
containTCP:= ContainSubstring(
		"udp local %s:%d external %s:%d vrf 0  out2in-only", 
		linuxTap2IP, udpSvcLocalPort, udpSvcExtIP, udpSvcExtPort
)
containUDP := // ...
```

with these we can have clear assertions:

checkConfig(true)
// vs.
Expect(staticMappings()).To(And(containTCP,	containUDP))
```

The `Expect(staticMappings())` actually checks for error first even though it returns output as well.


This also allows us to different TCP/UDP combinations for the assert.